data_model/src/param.rs

```diff
@@ -0,0 +1,246 @@
+    /// Limits that a smart contract must obey at runtime to not be considered invalid.
+    #[derive(
+        Debug,
+        Display,
+        Clone,
+        Copy,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        CopyGetters,
+        Decode,
+        Encode,
+        Serialize,
+        Deserialize,
+        IntoSchema,
+    )]
+    #[display(fmt = "{fuel},{memory}_SL")]
+    #[getset(get_copy = "pub")]
+    pub struct SmartContractLimits {
+        /// Maximum amount of fuel that a smart contract can consume
+        pub fuel: NonZeroU64,
+        /// Maximum amount of memory that a smart contract can use
+        pub memory: NonZeroU64,
+    }
+
+    /// Single blockchain [`Parameter`]. Controls how blockchain works.
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        EnumDiscriminants,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    #[ffi_type(opaque)]
+    pub enum Parameter {
+        /// Maximal amount of time a peer will wait before forcing creation of a new block.
+        ///
+        /// A block is created if this limit or `Self::TransactionsInBlock` limit is reached,
+        /// whichever comes first. Regardless of the limits, an empty block is never created.
+        BlockTime(Duration),
+        /// How long a peer will wait for a block to be committed.
+        ///
+        /// If this period expires the block will request a view change
+        CommitTime(Duration),
+        /// Block limits
+        BlockLimits(BlockLimits),
+        /// Transaction limits.
+        TransactionLimits(TransactionLimits),
+        /// Smart contract limits (user submitted smart contracts and triggers)
+        SmartContractLimits(SmartContractLimits),
+        /// Executor limits
+        ExecutorLimits(SmartContractLimits),
+        /// Blockchain specific parameter (defined in the executor)
+        Custom(JsonString),
+    }
+
+    /// Set of all current blockchain parameter values
+    ///
+    /// See [`Parameter`] for details
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    #[allow(missing_docs)]
+    pub struct Parameters {
+        pub block_time: Duration,
+        pub commit_time: Duration,
+        pub block_limits: BlockLimits,
+        pub transaction_limits: TransactionLimits,
+        pub executor_limits: SmartContractLimits,
+        pub smart_contract_limits: SmartContractLimits,
+        /// Collection of all custom parameters
+        pub custom: JsonString,
+    }
+}
+
+impl core::fmt::Display for Parameter {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Self::BlockTime(v) | Self::CommitTime(v) => core::fmt::Debug::fmt(&v, f),
```

Comment:
While with other parameters their display will obviously tell what the parameter is (e.g. `BlockLimits` -> `BlockLimits`), `Duration` would be confusing (is it `BlockTime` or `CommitTime`?) 

I would add a prefix here.